extensions/wikia/PortableInfobox/controllers/PortableInfoboxBuilderController.class.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+class PortableInfoboxBuilderController extends WikiaController {
+	const EXTENSION_PATH = 'extensions/wikia/PortableInfobox/';
+	const INFOBOX_BUILDER_TEMPLATE_PATH = 'templates/PortableInfoboxBuilderIndex.mustache';
+	const INFOBOX_BUILDER_STYLES_PATH = 'styles/PortableInfoboxBuilder.scss';
+
+	/**
+	 * creates portable infobox builder UI
+	 */
+	public function index() {
+		$this->optionsPlaceholder = wfMessage( 'portable-infobox-builder-edit-element-options-placeholder' )->text();
+		$this->publishBtn = wfMessage( 'portable-infobox-builder-publish-button' )->text();
+		$this->infoboxBuilderHeader = wfMessage( 'portable-infobox-builder-title' )->text();
+		$this->formAction = self::getLocalUrl( 'publish' );
+		$this->templateTitle = requestContext::getMain()->getTitle()->getPrefixedDBkey();
+		$this->infoboxContent = ( new PortableInfoboxRenderService() )
+			->renderInfobox( $this->getInfoboxData(), null, 'portable-infobox-layout-tabular' );
+
+		$this->response->getView()->setTemplatePath( $IP . self::EXTENSION_PATH . self::INFOBOX_BUILDER_TEMPLATE_PATH );
+		$this->response->setTemplateEngine( WikiaResponse::TEMPLATE_ENGINE_MUSTACHE );
+		$this->wg->out->addStyle(
+			AssetsManager::getInstance()->getSassCommonURL( self::EXTENSION_PATH . self::INFOBOX_BUILDER_STYLES_PATH )
+		);
+	}
+
+	/**
+	 * publishes changes to infobox template made inside infobox builder UI and redirects to template page
+	 */
+	public function publish() {
+		$title = $this->getContext()->getTitle();
+
+		$this->editInfoboxTemplate(
+			$title,
+			$this->getInfoboxMarkup(),
+			wfMessage( 'portable-infobox-builder-edit-summary' )->text()
+		);
+
+		$this->response->redirect( $title->getFullURL() );
+	}
+
+	/**
+	 * does edit on the infobox template
+	 *
+	 * @param Title $title
+	 * @param String $markup
+	 * @param String $summary
+	 *
+	 * @throws PermissionsError
+	 * @throws ReadOnlyError
+	 * @throws ThrottledError
+	 * @throws UserBlockedError
+	 *
+	 * @todo add error handling
```

Comment:
Im not sure if we are handling the $wgDisableAnonEditing when using WikiPage::doEdit - was it supported when using EditPage::attemptSave ?